app/models/base.rb

```diff
@@ -307,12 +307,29 @@ def self.parse_datacite_metadata(id: nil, response: nil)
     resource_type_general = attributes.dig("types", "resourceTypeGeneral")
     type = Bolognese::Utils::DC_TO_SO_TRANSLATIONS[resource_type_general.to_s.dasherize] # || attributes.dig("types", "schemaOrg")
 
+    registrant_id = client_id == "crossref.citations" ? get_crossref_member_id(id) : "datacite.#{client_id}"
+
     {
       "@id" => id,
       "@type" => type,
       "datePublished" => get_date(attributes["dates"], "Issued"),
       "proxyIdentifiers" => proxy_identifiers,
-      "registrantId" => "datacite.#{client_id}" }.compact
+      "registrantId" => registrant_id }.compact
+  end
+
+
+  def self.get_crossref_member_id(id, options={})
+    doi = doi_from_url(id)
+    return "crossref.citations" unless doi.present?
```

Comment:
Why are you returning `crossref.citations`?

Reply:
well in the context where get_crossref_member_id is used there is no need to do this check. But I added it in case anybody uses it the method in any other context.

Reply:
My problem is that this is triggered if no DOI is present. I don't see a use case for this.